pkg/registry/core/rest/storage_core.go

```diff
@@ -417,6 +417,12 @@ type componentStatusStorage struct {
 }
 
 func (s componentStatusStorage) serversToValidate() map[string]*componentstatus.Server {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DisableComponentStatusProbes) {
+		// This API was deprecated in 1.19 for the reasons explained in
+		// https://github.com/kubernetes/kubernetes/pull/93570
+		return map[string]*componentstatus.Server{}
+	}
```

Comment:
This doesn't really address my comment, so copying here again:

I disagree with this change.

API deprecation means that we won't be evolving it.
But GA APIs (and componentstatuses kind-of implicitly belongs to this category) can't be broken.
Not exposing this information has a potential to break some users and I'm definitely against doing this.

@thockin @deads2k - in absence of Jordan.

I would like to see wider ACK for this kind of changes.

Reply:
but this is very broken, that is my point
```
	serversToValidate := map[string]*componentstatus.Server{
		"controller-manager": {EnableHTTPS: true, TLSConfig: &tls.Config{InsecureSkipVerify: true}, Addr: "127.0.0.1", Port: ports.KubeControllerManagerPort, Path: "/healthz"},
		"scheduler":          {EnableHTTPS: true, TLSConfig: &tls.Config{InsecureSkipVerify: true}, Addr: "127.0.0.1", Port: kubeschedulerconfig.DefaultKubeSchedulerPort, Path: "/healthz"},
	}
```

I don't think we should have an API that makes these assumptions

Reply:
there is no rush for this, I just found it today and I'm fine to look for wider discussion, I will open an email thread in sig-arch

/hold

Reply:
100% agree it's a broken design, but it DOES work for some people.

AFAICT, this code is only triggered when GET/LIST is called, so there's no impact to having it (e.g. it doesn't periodically probe).

I don't object to commenting the code.

I don't even object to sending an API warning when it is accessed.

But do we need to do more than this?

Reply:
Yeah I agree with Tim. While this is broken in X% of cases, this change would effectively break 100% of cases. How is that better?

Reply:
What would make more sense to me is waiting for component-sli metrics to emerge in etcd, so that we have livez/readyz coverage across all k8s components in a readily ingestible state, then to put componentstatus behind a flag (i.e. it disappears unless you explicitly enable it) for 3 releases to give people ample time to properly migrate off of it and then wholesale remove it.